### Investigate binary variables assignment_attempt_number and assignment_max_attempts

In [1]:
%%capture

%run '../lib/libraries.ipynb'
from ..lib import utilities as util
from pyspark.sql import functions as F
import pandas as pd

In [2]:
filterDf = util.load_df('filterDf')

Appears binary but the variable name indicates it could have any values. The data only contains 1 and 0

In [3]:
dfPd = filterDf.select('assignment_attempt_number', 'assignment_max_attempts').toPandas()
pd.crosstab(dfPd.assignment_attempt_number.fillna('null'), dfPd.assignment_max_attempts.fillna('null'), margins=True, margins_name="Total")

assignment_max_attempts       0      1  Total
assignment_attempt_number                    
0                          1566      0   1566
1                             0  63272  63272
Total                      1566  63272  64838

- both variables only contain "0" 1566 out of 78,982 observations, otherwise the value is always "1"
- are they the same 1566 as the "[unassigned]" in response_correctness

###### Verify same variables as the  1566 as the "[unassigned]" in response_correctness

In [4]:
filterDf.where( ( F.col('response_correctness') == '[unassigned]') & (F.col('assignment_max_attempts') == 0) & (F.col('assignment_attempt_number') == 0)).select().count()

1566

#### Conclusion
- This is related to the 1566 as the "[unassigned]" in response_correctness
- ~~Address in the investigation of the 1566~~
- No action required
  - variables won't be used

In [5]:
util.finish_todo("Investigate binary variables assignment_attempt_number and assignment_max_attempts")

Finished: Investigate binary variables assignment_attempt_number and assignment_max_attempts
